### PR TITLE
Cryokinesis and Geladikinesis mutations

### PR DESCRIPTION
**Pull request by Time-Green**
Adds cryokinesis in the spirit of goon. It allows you to shoot a beam that freezes people and makes floors slippery.
Adds geladikinesis, gives you the ability to make snow and snow-structures. 

Also fixed snow walls runtiming like fucking crazy because they're treated like objects in crafting

Also added snow tiles to snow crafting

:cl:
add: Adds the cryokinesis and geladikinesis mutations
fix: Snow walls no longer runtime everything
/:cl:
geladi means frozen. I swear im not a frozen fan bit nixkinesis doesnt sound that great
Why: Cryokinesis is fun to mess around with, and geladikinesis is great if you want a bunch of assistants turning dorms into Siberia

### CODE REVIEW

File: code/datums/mutations/cold.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/datum/mutation/human/geladikinesis
+	name = "Geladikinetic Creation"
+	desc = "Allows the user to concentrated moisture and sub-zero forces into snow."
```
Comment (ATH1909) [inline code/datums/mutations/cold.dm]: Replaced "concentrated" with "concentrate".

**Comment by ATH1909** [comment]
Wait, is this a stealth temperature gun buff (or nerf? is the new way the cold beams work less or more effective than the old way?)?

Hey, I'm not complaining, but you should probably mention that in the PR.

File: code/__DEFINES/flags.dm [hunk 2]
```diff
@@ -1,162 +1,162 @@
...
-#define MOBILITY_FLAGS_DEFAULT (MOBILITY_MOVE | MOBILITY_STAND | MOBILITY_PICKUP | MOBILITY_USE | MOBILITY_UI | MOBILITY_STORAGE | MOBILITY_PULL)
-#define MOBILITY_FLAGS_INTERACTION (MOBILITY_USE | MOBILITY_PICKUP | MOBILITY_UI | MOBILITY_STORAGE)
-
-// radiation
-#define RAD_PROTECT_CONTENTS (1<<0)
-#define RAD_NO_CONTAMINATE (1<<1)
-
-//alternate appearance flags
-#define AA_TARGET_SEE_APPEARANCE (1<<0)
+/*
```
Comment (Cyberboss) [inline code/__DEFINES/flags.dm]: Revert the line ending changes

File: code/modules/projectiles/projectile/special/temperature.dm [hunk 3]
```diff
@@ -16,3 +16,14 @@
...
 	name = "heat beam"
 	temperature = 400
+
+/obj/item/projectile/temp/cryo
+	name = "cryo beam"
+	range = 3
+
+/obj/item/projectile/temp/cryo/on_range()
+	var/turf/T = get_turf(src)
+	if(istype(T, /turf/open))
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/special/temperature.dm]:
```suggestion
	if(isopenturf(T))
```

File: code/modules/spells/spell_types/conjure.dm [hunk 4]
```diff
@@ -80,9 +80,10 @@
 	school = "conjuration"
 	charge_max = 150
 	cooldown_min = 10
+	var/delete_old = TRUE //TRUE to delete the last summoned object if it's still there, FALSE for infinite item stream weeeee
 
 /obj/effect/proc_holder/spell/targeted/conjure_item/cast(list/targets, mob/user = usr)
-	if (item && !QDELETED(item))
+	if (delete_old && item && !QDELETED(item))
 		qdel(item)
 		item = null
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/conjure.dm]:
```suggestion
```

File: code/modules/spells/spell_types/conjure.dm [hunk 5]
```diff
@@ -80,9 +80,10 @@
 	school = "conjuration"
 	charge_max = 150
 	cooldown_min = 10
+	var/delete_old = TRUE //TRUE to delete the last summoned object if it's still there, FALSE for infinite item stream weeeee
 
 /obj/effect/proc_holder/spell/targeted/conjure_item/cast(list/targets, mob/user = usr)
-	if (item && !QDELETED(item))
+	if (delete_old && item && !QDELETED(item))
 		qdel(item)
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/conjure.dm]:
```suggestion
		QDEL_NULL(item)
```

File: code/game/objects/items/stacks/stack.dm [hunk 6]
```diff
@@ -191,9 +191,15 @@
 		var/obj/O
 		if(R.max_res_amount > 1) //Is it a stack?
 			O = new R.result_type(usr.drop_location(), R.res_amount * multiplier)
+		else if(ispath(R.result_type, /turf/))
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/stack.dm]:
```suggestion
		else if(ispath(R.result_type, /turf))
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/__DEFINES/flags.dm (hunk 2 above)
Comment (Time-Green) [reply code/__DEFINES/flags.dm]: how would I do that? I've reset the file to master and nothing changed

Comment (Cyberboss) [reply code/__DEFINES/flags.dm]: https://stackoverflow.com/a/11341700/3976486

**Comment by Time-Green** [comment]
Thanks. I unfucked it

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**